### PR TITLE
RHOAIENG-16517: chore(tests): add new changed-files-rebuild helper for GitHub Actions PRs

### PR DESCRIPTION
**Pull request by jiridanek**
https://issues.redhat.com/browse/RHOAIENG-16517

## Description

Small Go script that analyzes a Dockerfile and prints what's required to build it.

This is then running from the Python script to find out what Dockerfiles need a rebuild.

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by jiridanek** [comment]
/override ci/prow/images

same old, same old

File: scripts/buildinputs/go.mod [hunk 1]
```diff
@@ -0,0 +1,60 @@
+module dockerfile
+
+go 1.22.0
```
Comment (jstourac) [inline scripts/buildinputs/go.mod:3]: This is okay, just checking why not 1.23 ? It is different to what we use in odh/kubeflow anyway. Or different reason?

**Review by jstourac — COMMENTED** [review]
Thank you for this addition. I didn't test it on my end anyway. I'm not a golanger so I could miss something, but let's move this forward if it is working somehow and we can improve in the future.

/lgtm

File: scripts/buildinputs/buildinputs_test.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+			files = append(files, path)
+		}
+		return nil
+	})
+
+	return files, err
+}
+
+// TestParseAllDockerfiles checks there are no panics when processing all Dockerfiles we have
+func TestParseAllDockerfiles(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (jiridanek) [reply scripts/buildinputs/buildinputs_test.go:23]: I expected to run this by hand, as needed. Same as the other tests for the scripts in the ci/ directory that also don't run automatically.

File: scripts/buildinputs/go.mod (hunk 1 above)
Comment (jiridanek) [reply scripts/buildinputs/go.mod:3]: No particular reason, anything > 1.21 should be fine, I guess I had 1.22 around so that's what ended up here.

**Comment by jiridanek** [comment]
/approve
thanks!

**Comment by jiridanek** [comment]
>error: build error: Failed to push image: trying to reuse blob sha256:d1626b5b9f7a59add02f7c14fb670e63a729a8b70a53b95f56bf7f5ee8dc12d8 at destination: unable to retrieve auth token: invalid username/password: authentication required

thanks but no thanks

/override ci/prow/images

**Pull request merged**